src/Query.php

```diff
@@ -94,7 +98,7 @@ public function findMany(array $ids): Collection
             return collect();
         }
 
-        if (($results = static::recordCache()->only($ids)) && count($ids) === $results->count()) {
+        if (($results = static::recordCache()->only($ids)->values()) && count($ids) === $results->count()) {
```

Comment:
Perhaps we could first check to see if all the values exist, and then intersect the array to get the values?

Reply:
This implementation has been simplified using `documentCache()->has($ids)` and then iterating the ids to retrieve them from the cache with `documentCache()->get($id)`.

Now the order of ids passed `findMany($ids)` is the order of the Collection returned.